apps/tunes/management/commands/tunes_clear_duplicate_songs_from_database.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+from django.conf import settings
+from django.db.models.signals import post_save
+
+from accounts.signals import create_user_emotion_records
+from base.management.commands import MoodyBaseCommand
+from tests.helpers import SignalDisconnect
+from tunes.models import Song
+
+
+class Command(MoodyBaseCommand):
+    help = 'Clear duplicate songs from the database'
+
+    def __init__(self, *args, **kwargs):
+        super(Command, self).__init__(*args, **kwargs)
+        self.logger = logging.getLogger(__name__)
+
+    def get_duplicate_songs_for_song(self, song):
+        """Return all songs in the database that are duplicates of the specified song"""
+        return Song.objects.filter(
+            name=song.name,
+            artist=song.artist
+        ).exclude(
+            code=song.code
+        ).order_by(
+            'created'
+        )
+
+    def reassign_votes_for_dupe_song_to_canonical_song(self, canonical_song, dupe_song):
+        """
+        Reassign all votes for the duplicate song to the canonical song
+        If a user has voted on both the canonical song and duplicate song, only keep the vote for the canonical song
+        :param canonical_song: (Song) Song we will persist in our database for a given name and artist
+        :param dupe_song: (Song) Song marked as a duplicate of the canonical song
+        """
+        dupe_votes = dupe_song.usersongvote_set.count()
+
+        for vote in dupe_song.usersongvote_set.iterator():
+            if vote.user.usersongvote_set.filter(song=canonical_song).exists():
+                vote.delete(update_boundaries=False)
+            else:
+                vote.song = canonical_song
+
+                # Disable signal to update user emotion boundaries
+                dispatch_uid = 'user_post_save_create_useremotion_records'
+                with SignalDisconnect(post_save, create_user_emotion_records, settings.AUTH_USER_MODEL, dispatch_uid):
```

Comment:
We're disabling the wrong signal; we should disable the `update_user_emot_boundaries` post_save signal  on UserSongVote